src/js/common/utils/accessible-menus.js

```diff
@@ -340,10 +361,32 @@ export default function addMenuListeners(menuElement) {
     }
   });
 
-  // TODO: Implement this and get rid of MegaMenu usage
-  // menuElement.addEventListener('click', (e) => {
-  //   // Handle opening menus
-  //   // Handle closing menus when clicked away
-  // });
+  // Close when the screen is resized
+  // Useful for when the menu might be hidden on smaller screens (nav menubar)
+  if (closeOnResize) {
+    window.addEventListener('resize', () => closeAll(menuElement));
+  }
+
+  menuElement.addEventListener('click', e => {
+    const targetLi = e.target.parentElement;
+    // Target's grandparent because the parent is a <li>
+    const inMenubar = targetLi.parentElement.getAttribute('role').toLowerCase() === 'menubar';
+    const inMenu = targetLi.parentElement.getAttribute('role').toLowerCase() === 'menu';
+
+    // Handle opening (and closing) menus
+    if (inMenubar) {
+      toggleMenu(targetLi);
+    } else if (inMenu) {
+      openMenu(targetLi);
+    }
+  });
+
+  // Handle clicking away from the menu
+  document.addEventListener('click', event => {
+    const target = event.target;
+    if (!menuElement.contains(target)) {
```

Comment:
This line got me thinking about how we might want to allow users to open menus via hover, also if the user clicks on a part of the menu that is not a li, the menu closes, which is an edge case but which doesn't seem like expected behavior. Shall we include these in the enhancements ticket?

Reply:
I don't now if opening menus via hover is the desired functionality (enter UX researchers!), but if it is, we could give it a shot. Probably should write a ticket to do some discovery on that.

For clicking on a part of the menu that isn't an `li`, do you mean a menu item that isn't in an `li`, or like the space between / around / above / below the menu items but still "in" the menu?